splunk-otel-android/src/main/java/com/splunk/rum/RuntimeDetailsExtractor.java

```diff
@@ -22,44 +22,69 @@
 import android.content.IntentFilter;
 import android.os.BatteryManager;
 import androidx.annotation.Nullable;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
 import java.io.File;
 
 /** Represents details about the runtime environment at a time */
-final class RuntimeDetails extends BroadcastReceiver {
+final class RuntimeDetailsExtractor<RQ, RS> extends BroadcastReceiver
```

Comment:
I left this entire class in the Splunk part of the codebase -- OTel will probably want to export this stuff as metrics, not span attributes.